### ISSUE TITLE
Accessibility to Training Data

### ISSUE THREAD

**Issue by Daisy-GENG** [roahmlab/terrain_friction_dataset#1]
Hi, thanks for your amazing work!

I noticed that there is only the friction label and the corresponding rgb image in this repo. Is it possible to offer the whole dataset, including training, validation as well as test sequences?

Thanks in advance.

Best regards,
Daisy

**Comment by ParkerEwen5441** [comment]
Hi Daisy,

This dataset was only used to estimate a probabilistic model of the
friction for each terrain class. The RGB image is used to indicate the type
of terrain class corresponding to each label but is not used during model
generation.

In our paper we mention that we use a Gaussian distribution to model the
friction of each terrain class and use this dataset to compute the mean and
variance of the distribution.

If you have any other questions I'd be happy to answer them!

Cheers,
Parker

**Comment by Daisy-GENG** [comment]
Hi Parker,

Thanks for your answer. I am afraid that I have not made myself clear. I want to know how you get the rgbd training data for terrain type segmentation intead of terrain property estimation. Is it possible for you to publicize the raw rgbd images as well as their terrain type semantic labels.

Another question is that how you make sure that your acceleartion is 0 when measuring the pulling force, because it seems that you assume that pulling force is equivalent to the friction force in the paper. And how you make sure that the pulling force is  horizontal which is vertical to the gravity direction? It will be very nice if you can offer an experiment setup picture.

Thank you so much!

Best,
Daisy

**Comment by ParkerEwen5441** [comment]
Hi Daisy,

For the terrain segmentation we used this work:
https://github.com/CSAILVision/semantic-segmentation-pytorch

The dataset used to train the semantic segmentation network is provided in
this repo and we relabel some of the classes (ie. Road->concrete) for our
purposes. You can also change the semantic segmentation network used to one
you are more familiar with. I hope this helps!

Cheers,
Parker

**Comment by ParkerEwen5441** [comment]
I can send an experiment setup image later today

**Comment by Daisy-GENG** [comment]
Thank you so much for your detailed answer!

Best,
Daisy

**Comment by Daisy-GENG** [comment]
Hi, is it possible for you to send me an image of your experiment setup? My email is rgengaa@connect.ust.hk

Thanks a lot!

Best,
Daisy

**Comment by zhj3180** [comment]
Hello, I recently read your paper and would like to try replicating the experiment. Could you please send me an image of the experiment setup? Thank you very much for your help.   hjzhang0831@bit.edu.cn

**Comment by ParkerEwen5441** [comment]
Hi zhj3180! I've attached a picture of the experimental setup. We used a weighted device with a rubber pad on the bottom (same material as the rubber feet of legged robots, shown in blue) and attached a strain gauge to the top (shown in red). We pulled this device across the surfaces and measured the force of the strain gauge and used the mass of the weighted device device to compute the coefficient of friction.

While its quite rudimentary, we recently validated the friction measurements from this experimental setup using a more accurate force-torque sensor and found them to be approximately the same.

![friction](https://github.com/roahmlab/terrain_friction_dataset/assets/48282126/47c0faa8-c224-4bfb-a1b8-c1fe88fea2c4)

**Comment by zhj3180** [comment]
thanks a lot for your help!
